.github/workflows/rust.yml

```diff
@@ -40,6 +40,9 @@ jobs:
     # Check the submitted change meets style guidelines
     - name: Cargo Format
       run: cargo fmt --check
+    # Check the submitted change passes the clippy linter
```

Comment:
It might be worth making a new job for this, as it adds like 30s to our build. Then it will run in parallel on another build agent.